Intel ARC support

TotalDay opened the pull request:
Add OpenCL support

intel_extension_for_pytorch-1.13.10%2Bxpu-cp310-cp310-linux_x86_64.whl
torch-1.13.0a0%2Bgitb1dde16-cp310-cp310-linux_x86_64.whl

https://github.com/TotalDay/Intel_ARC_GPU_WSL_Stable_Diffusion_WEBUI

vladmandic committed:
Merge pull request #251 from wk5ovc/patch-3

Fix #8984

vladmandic committed:
Merge pull request #250 from Sakura-Luna/pre-fix

Preview fix

vladmandic committed:
Merge pull request #249 from gaojie111119/fix-noattr-elem_classes

fix #9234 #8970 check attr "elem_classes" before use

vladmandic committed:
Merge pull request #248 from GoulartNogueira/master

fix image orientation

vladmandic committed:
Merge pull request #245 from GeorgLegato/get_uiCurrentTab_Gr3.23_wrong

Update script.js

vladmandic committed:
Merge pull request #244 from Pluventi/master

Update postprocessing.py

vladmandic committed:
Merge pull request #241 from wywywywy/bug-outpainting-mk2-file-format

Bug outpainting mk2 file format

vladmandic committed:
Merge pull request #237 from Z-nonymous/master

Fix #9185

vladmandic committed:
Merge pull request #233 from Yan233th/master

Fix hasattr to in method

vladmandic committed:
Merge pull request #232 from space-nuko/improve-custom-code-extension

Improve custom code extension

vladmandic committed:
Merge pull request #231 from muerrilla/dropdown-dup-fix

Dropdown CSS fix

vladmandic committed:
Merge pull request #228 from Vespinian/fix_img2img_scriptrunner_for_gui

Changed img2img scriptrunner for gui request from scripts_txt2img to …

vladmandic committed:
update setup

vladmandic committed:
post merge fixes

vladmandic committed:
update webui.sh

vladmandic committed:
disable xformers/sdp if cannot be used

vladmandic committed:
fix dtype logic

vladmandic committed:
fix parsing of requirements

vladmandic committed:
add builtin gradio themes

gitadmin0608 committed:
CFGDenoiser and script_callbacks mod for SAG

vladmandic committed:
Create FUNDING.yml

nonnonstop committed:
Fix problem when --data-dir is specified

vladmandic committed:
add exception around torch properties

vladmandic committed:
fix lora/lyco dir

vladmandic committed:
add setup.log note

vladmandic committed:
Merge pull request #366 from nonnonstop/fix-datadir

Fix problems when --data-dir is specified

vladmandic committed:
Merge pull request #352 from ashen-sensored/master

CFGDenoiser and script_callbacks modification for SAG

vladmandic committed:
update setup to deal with extensions preload

vladmandic committed:
cleanup setup

vladmandic committed:
update extension

vladmandic committed:
fix lora dir

vladmandic committed:
extend system paths

vladmandic committed:
Merge pull request #383 from missionfloyd/extra-network-none

Add "None" option to extra networks dropdowns

vladmandic committed:
disable python 3.11

vladmandic committed:
update

vladmandic committed:
fix model_name

vladmandic committed:
make tensorflow optional

vladmandic committed:
update lora logic

vladmandic committed:
Merge branch 'master' into master

vladmandic committed:
Merge pull request #385 from AlUlkesh/master

fix: lightboxModal, selectedTab

vladmandic committed:
update post merge

vladmandic committed:
apply patch

vladmandic committed:
fix lightbox

vladmandic committed:
handle torch cuda crashing in setup

vladmandic committed:
update swinir

missionfloyd committed:
Merge branch 'master' into extras-sliders

pangbo13 committed:
Merge branch 'master' into xyz-plot-dropdown-vladmandic

one111eric committed:
fix typo

one111eric committed:
fix for vlad fork

Sakura-Luna committed:
Merge branch 'master' into pr9295

vladmandic committed:
combo patch

vladmandic committed:
update argparse

vladmandic committed:
Merge pull request #438 from Sakura-Luna/pr9295

Add bf16 support for VAE

vladmandic committed:
premerge cleanup

vladmandic committed:
Merge pull request #435 from pangbo13/xyz-plot-dropdown-vladmandic

Add dropdown for X/Y/Z plot

emirhnergn committed:
invert sampler selection process #447

emirhnergn committed:
Merge branch 'master' of https://github.com/vladmandic/automatic

vladmandic committed:
add token merging and vae rollback

vladmandic committed:
Merge pull request #452 from emirhnergn/master

Invert sampler selection process

vladmandic committed:
fix sampler list

vladmandic committed:
increase timeout

lshqqytiger committed:
First DirectML implementation.
Unstable and not tested.

lshqqytiger committed:
Fix PLMS & DPM & DDIM.
Unstable and not tested.

lshqqytiger committed:
Update TODO_DML.

ilikenwf commented:
I don't have an ARC gpu, but out of curiosity, what kind of performance are you seeing?

vladmandic committed:
add rebase

vladmandic committed:
update samplers

vladmandic committed:
update vae

vladmandic committed:
update gpu util search paths

vladmandic committed:
disable git status check

vladmandic committed:
increase git debug logging

vladmandic committed:
add compatibility opts

lshqqytiger committed:
Add hijack for DirectML.
Unstable & Under testing.

vladmandic committed:
ti fixes

vladmandic committed:
update geninfo

lshqqytiger committed:
need full precision for model & vae.
Stable & tested.

lshqqytiger committed:
fix

lshqqytiger committed:
fix

lshqqytiger committed:
Add dml_specific.

lshqqytiger committed:
Merge branch 'master' into directml

lshqqytiger committed:
Merge branch 'master' into directml

DrakeRichards committed:
#488: Added check for initial tmp image

DrakeRichards committed:
Refined regexp to not check the path

vladmandic committed:
fix api image paths

Gazzoo-byte committed:
Change 'Automatic' behaviour

Modifies the behaviour of 'Automatic' to check in vae_path for vae files with a filename that matches the checkpoint, failing this falls back to None

vladmandic committed:
fix embedding preview

vladmandic committed:
Merge pull request #498 from Gazzoo-byte/patch-2

Change 'Automatic' behaviour

Gazzoo-byte committed:
Fix a bug

Forgot a return value which was causing 'ValueError: too many values to unpack (expected 2)' upon trying load a VAE with Automatic in sd_vae

vladmandic committed:
Merge pull request #502 from Gazzoo-byte/patch-3

Fix ValueError: too many values to unpack (expected 2)

DrakeRichards committed:
Further refined regexp pattern

Gazzoo-byte committed:
sd_vae automatic behaviour fix for non-Windows users

Hadn't considered that hard-coding the separators would break this for non-Windows users - replaced hard coding with os.sep

Gazzoo-byte committed:
Update sd_vae.py

vladmandic committed:
validate script_args

Gazzoo-byte committed:
Update sd_vae.py

Removed pathlib, but had to retain '+ ".vae.pt"' etc, otherwise it was adding a separator between the filename and the extension

vladmandic committed:
Merge pull request #510 from Gazzoo-byte/patch-4

sd_vae automatic behaviour fix for non-Windows users

vladmandic committed:
cleanup vae messages

vladmandic committed:
Merge pull request #495 from DrakeRichards/vladmandic-notifications-fix

Fix multiple notifications on refresh

vladmandic committed:
reorder setup items

lshqqytiger committed:
Implement torch.dml.
VERY UNSTABLE & NOT TESTED.

lshqqytiger committed:
fix

lshqqytiger committed:
fix.
Unstable & need more test.

lshqqytiger committed:
Implement memory estimation for AMDGPUs.
Stable.

lshqqytiger committed:
Move realesrgan fallback to dml/hijack.

vladmandic committed:
add compatibility cmdflags

lshqqytiger committed:
Merge branch 'master' into directml

vladmandic committed:
update rollback vae

Scholar01 committed:
support for macOS.

vladmandic committed:
cleanup scripts

vladmandic committed:
increase thumnails

vladmandic committed:
fix geninfo pretty print

vladmandic committed:
add shared.url

vladmandic committed:
Merge pull request #521 from Scholar01/mac

support for macOS.

vladmandic committed:
add asyncio handler

one111eric committed:
remove test output

vladmandic committed:
fix realesrgan path, secondary sampler

vladmandic committed:
Merge pull request #437 from one111eric/dev-add-extra-network-description-vlad

add extra network description

vladmandic committed:
Merge pull request #235 from missionfloyd/extras-sliders

Extras sliders

vladmandic committed:
reduce extra network exceptions

cool-bigdogs-tshirt committed:
attempt at unipc latent upscaling

i should have taken linear algebra before i dropped out...

cool-bigdogs-tshirt committed:
xyz override for latent upscaler fallback

vladmandic committed:
jumbo patch

vladmandic committed:
update

derVedro committed:
Update webui.sh

webui can launch from any directory

ancker010 committed:
Renable Zip button in black-orange.css

lshqqytiger committed:
torch 2.0.0 support.

vladmandic committed:
update models_dir

vladmandic committed:
Merge pull request #577 from ancker010/master

Renable Zip button in black-orange.css

vladmandic committed:
Merge pull request #576 from derVedro/patch-1

webui.sh launches from any directory

vladmandic committed:
handle clip_skip is none

vladmandic committed:
configurable train log

vladmandic committed:
update settings

vladmandic committed:
cleanup scripts

vladmandic committed:
update cli requirements

Loufe commented:
You're effectively trying to PR the entire Vladmandic repo into this one. You should try setting this PR up again.

vladmandic committed:
update todo

DrakeRichards committed:
Moved exra network close button

vladmandic committed:
Merge pull request #601 from DrakeRichards/extra-network-ui-adjustments

Moving Extra Network close button

DrakeRichards committed:
Added notification.mp3 support

vladmandic committed:
update options

cool-bigdogs-tshirt committed:
unipc img2img - add a bunch of code to get a single value that maybe performs slightly better?

cool-bigdogs-tshirt committed:
fix batching issue

vladmandic committed:
change order of argparse

vladmandic committed:
Merge pull request #568 from cool-bigdogs-tshirt/bigdogs/unipc-latent-upscale

unipc latent upscale

vladmandic committed:
fix prompts from file

vladmandic committed:
fix cards previews

vladmandic committed:
xyz improvements

AuwowaUwU committed:
Remove hardcoded disabling of grids in file script

vladmandic committed:
Merge pull request #620 from AuwowaUwU/patch-1

Remove hardcoded disabling of grids in file script

lshqqytiger committed:
Merge branch 'master' into directml

lshqqytiger committed:
Load libatiadlxx.so on Linux systems.

lshqqytiger committed:
Sync submodules.

lshqqytiger committed:
Sync submodules.

vladmandic committed:
fix awalys on script args

vladmandic committed:
critical fix

vladmandic committed:
fix extra networks

nekoworkshop committed:
Initial implementation

vladmandic committed:
make clip skip persisent

dpinav committed:
Fix typo that prevents training Textual Inversion

There was a small typo in line 529.

shared.ops.embeddings_train_log caused an attribute not found exception when training TIs.

nekoworkshop committed:
Use SharedSettingsStackHelper

vladmandic committed:
Merge pull request #628 from dpinav/master

Fix typo that causes exception when training Textual Inversion

vladmandic committed:
add memory monitor

vladmandic committed:
update repos

nekoworkshop committed:
Adjust axis names to be shorter.

nekoworkshop committed:
Extra information in ToMe related settings

vladmandic committed:
force lightbox

vladmandic committed:
update lint rules

vladmandic committed:
add test mode

vladmandic committed:
set git triggers

vladmandic committed:
critical

vladmandic committed:
add debug mode

vladmandic committed:
update cmdflags

lshqqytiger committed:
Move/Remove hijacks. Add warning.

lshqqytiger committed:
Approx NN works fine.

lshqqytiger committed:
Merge pull request #622 from vladmandic/directml

Add DirectML backend.

nekoworkshop committed:
Remove less useful ToMe options from the xyz plot

Disty0 committed:
Intel ARC Support

Disty0 committed:
Revert force cross_attention_optimization

vladmandic committed:
update ti folders

Disty0 committed:
More patches and less import shared

vladmandic committed:
Merge pull request #632 from nekoworkshop/dev

Adding ToMe parameters to available x/y/z plot options

vladmandic committed:
update logging

vladmandic committed:
add dynamo options

vladmandic committed:
new command line parser

Disty0 committed:
More patches and Import IPEX after Torch

Disty0 committed:
Merge remote-tracking branch 'upstream/master'

Disty0 committed:
undo cli

Disty0 committed:
undo cli

vladmandic committed:
Merge pull request #643 from Disty0/master

Intel ARC Support

vladmandic committed:
update todo

vladmandic committed:
cleanup installer

vladmandic committed:
remove circular imports from installer

cool-bigdogs-tshirt committed:
fix unipc img2img denoising sample count

was wrongly using the inverse of the intended value. smaller denoising
strength should run fewer steps.

vladmandic committed:
Merge pull request #653 from cool-bigdogs-tshirt/bigdog/unipc-img2img-steps

fix unipc img2img denoising sample count

vladmandic committed:
update

vladmandic committed:
fix script api

vladmandic committed:
fix embedding logging

vladmandic committed:
fallback args

vladmandic committed:
fix argparse

vladmandic committed:
fix lora memory leak

vladmandic committed:
update requirements

DrakeRichards committed:
Merge branch 'master' of https://github.com/vladmandic/automatic into notification-sounds

vladmandic committed:
update requirements

DrakeRichards committed:
Added notification sound settings

vladmandic committed:
update requirements

vladmandic committed:
switch cmdargs

lshqqytiger committed:
remove NotImplementedError.

DrakeRichards committed:
Merge branch 'master' into notification-sounds

vladmandic committed:
update

vladmandic committed:
cleanup

vladmandic committed:
fix exif data handler

vladmandic committed:
redesign logging

vladmandic committed:
merge commits

vladmandic committed:
Merge pull request #606 from DrakeRichards/notification-sounds

Adding notification.mp3 support

vladmandic committed:
fix logger

vladmandic committed:
fix logger

scottmudge committed:
fix issue with gradio UI not loading when google fonts API is not accessible

vladmandic committed:
Merge pull request #712 from scottmudge/fix-offline-font-access

Fix issue with gradio UI not loading when google fonts API is not accessible

vladmandic committed:
optimize model load

cool-bigdogs-tshirt committed:
adjust unipc img2img parameters

vladmandic committed:
fonts and upscale fix

vladmandic committed:
update ssl

vladmandic committed:
update requirements

vladmandic committed:
fix image temp files

vladmandic committed:
Merge pull request #721 from cool-bigdogs-tshirt/bigdog/more-unipc-experiments

bigdog/more unipc experiments

vladmandic committed:
set default gallery view

Disty0 committed:
Add IPEX Optimizers and use XPU instead of CPU when using IPEX

vladmandic committed:
add version flag

Disty0 committed:
Use cmd_args parser instead of launch.py

vladmandic committed:
add notifications

vladmandic committed:
Merge pull request #728 from Disty0/master

Add IPEX Optimizers and use XPU instead of CPU when using IPEX

vladmandic committed:
restart server redesign

vladmandic committed:
minor formatting updates

vladmandic committed:
fix notifications

vladmandic committed:
update gitignore

vladmandic committed:
add ignore flag

Disty0 committed:
Remove unnecessary IPEX imports

vladmandic committed:
Merge pull request #750 from Disty0/master

Remove unnecessary IPEX imports

vladmandic committed:
merge from upstream

vladmandic committed:
draft full merge from upstream

vladmandic committed:
add api-logo workaround

vladmandic committed:
merge fixes

Disty0 committed:
Fix memory monitoring when using IPEX

vladmandic committed:
Merge pull request #768 from Disty0/master

Fix memory monitoring when using IPEX

vladmandic committed:
add train preprocess options

vladmandic committed:
readd api docs

vladmandic committed:
lora messages

vladmandic committed:
update caption filename logic

vladmandic committed:
update

wbh1129 committed:
use proper shebangs in cli scripts

wbh1129 committed:
ui_extra_networks.py: make description default to "" instead of error

vladmandic committed:
Merge pull request #786 from wbh1129/extra_networks_description

ui_extra_networks.py: make description default to "" instead of error

vladmandic committed:
Merge pull request #782 from wbh1129/usrbinenv

Use proper shebangs in cli scripts

AUTOMATIC1111 commented:
Please examine the "files changed" tab of PRs you are creating. I think I've already seen two that try to pull the entire fork into the main repo.
